lib/Service/RecipeService.php

```diff
@@ -760,59 +775,12 @@ public function addRecipe($json) {
 
 			// The image field was empty, remove images in the recipe folder
 		} else {
-			if ($recipe_folder->nodeExists('full.jpg')) {
-				$recipe_folder->get('full.jpg')->delete();
-			}
-
-			if ($recipe_folder->nodeExists('thumb.jpg')) {
-				$recipe_folder->get('thumb.jpg')->delete();
-			}
-
-			if ($recipe_folder->nodeExists('thumb16.jpg')) {
-				$recipe_folder->get('thumb16.jpg')->delete();
-			}
+			$this->imageService->dropImage($recipe_folder);
 		}
 
 		// If image data was fetched, write it to disk
 		if ($full_image_data) {
-			// Write the full image
-			try {
-				$full_image_file = $recipe_folder->get('full.jpg');
-			} catch (NotFoundException $e) {
-				$full_image_file = $recipe_folder->newFile('full.jpg');
-			}
-
-			$full_image_file->putContent($full_image_data);
-
-			// Temp file to load data from
-			$tmp = tmpfile();
-			$tmpPath = stream_get_meta_data($tmp)['uri'];
-			fwrite($tmp, $full_image_data);
-			fflush($tmp);
-
-			// Write the thumbnail
-			$thumb_image = new Image();
-			$thumb_image->loadFromFile($tmpPath);
-			$thumb_image->fixOrientation();
-			$thumb_image->resize(256);
-			$thumb_image->centerCrop();
-
-			try {
-				$thumb_image_file = $recipe_folder->get('thumb.jpg');
-			} catch (NotFoundException $e) {
-				$thumb_image_file = $recipe_folder->newFile('thumb.jpg');
-			}
-
-			$thumb_image_file->putContent($thumb_image->data());
-
-			// Create low-resolution thumbnail preview
-			$low_res_thumb_image = $thumb_image->resizeCopy(16);
-			try {
-				$low_res_thumb_image_file = $recipe_folder->get('thumb16.jpg');
-			} catch (NotFoundException $e) {
-				$low_res_thumb_image_file = $recipe_folder->newFile('thumb16.jpg');
-			}
-			$low_res_thumb_image_file->putContent($low_res_thumb_image->data());
+			$this->imageService->setImageData($recipe_folder, $full_image_data);
```

Comment:
what about exceptions thrown here? (Currently `OCP\Files\NotFoundException`, `RecipeImageExistsException`, `GenericFileException`, `NotPermittedException`, and `LockedException`

Reply:
These are not handled in the current code as well. Once we have broken this class apart, we should be able to solve such problems.